x-pack/platform/plugins/shared/cloud/server/cloud_data/persist_token.ts

```diff
@@ -72,9 +81,10 @@ export const persistTokenCloudData = async (
           onboardingData: {
             solutionType:
               (solutionType as SolutionType) ?? cloudDataSo?.attributes.onboardingData.solutionType,
-            token: onboardingToken ?? cloudDataSo?.attributes.onboardingData.token,
+            token: onboardingToken ?? cloudDataSo?.attributes.onboardingData.token ?? '',
             security: security ?? securityAttributes,
           },
+          resourceData: resourceData ?? cloudDataSo?.attributes.resourceData,
```

Comment:
Do we need to do that here too?
```suggestion
          resourceData: resourceData ?? cloudDataSo?.attributes.resourceData ?? '',
```

Reply:
Talked to @galaxxyz, I am wrong here since resourceData is an object and the resourceData is optional there is no need to default it. Thanks @galaxxyz for talking to me about it.